### ISSUE TITLE
Scope is creeping. For example: should we even *have* rustfmt support?

### ISSUE THREAD

**Issue by chris-morgan** [rust-lang/rust.vim#298]
*I hear the soft note of the echoing voice of a creeping feature scope—*

Why don’t we throw away our rustfmt handling, and leave it to ALE and its rustfmt fixer?

Now I haven’t thought this through particularly hard, and I haven’t compared the results of them both (it’s probably a year since I last ran rustfmt); but ALE is *designed* to handle those sorts of things well, and much more flexibly than we are.

Let’s talk about the scope of this project, with reference to what other projects do, and especially what the Vim runtime is in the habit of doing (since rust.vim changes are now intended to be upstreamed there).

rustfmt is just the first thing that springs to mind, due to the various issues that have been coming up about it recently; I haven’t yet contemplated whether there are other parts that we should perhaps remove.

Feel free to now shout me down if you think that we should be including these sorts of things; bonus marks if your shouting incorporates research.

**Comment by dimbleby** [comment]
Happened to spot this while updating a merge request.  FWIW I'd have no objection to `rustfmt` being removed from this plugin; I think most ft plugins don't include a formatter.  (But most file types don't have a canonical formatter, so that's not exactly a knockout blow).

Suggest that all the syntastic stuff could go too.  IMO if such integration is required, then it belongs in syntastic.

**Comment by BurntSushi** [comment]
> Suggest that all the syntastic stuff could go too. IMO if such integration is required, then it belongs in syntastic.

I don't agree. IIRC, the syntastic stuff is in this repo because the maintainers of syntastic gave up maintaining Rust support. This happened years ago though.

**Comment by chris-morgan** [comment]
Yeah, ALE maintains its own Rust linters, and I think Rust is stable enough now that it’d be good to shift the Syntastic syntax checker from here to there.

**Comment by ulidtko** [comment]
FWIW, this plugin is *still* the only one which has `:RustFmtRange` sort-of working. vim-codefmt [didn't enable][1] the `--file-lines` support (for a decent reason, but didn't — even in a forward-compatible way). 
`vim-autoformat` [didn't merge][2] as well.

So uh... yeah. Talk about scope creep.

[1]: https://github.com/google/vim-codefmt/pull/129/files#diff-dde9132e1ffb539ce1443dc7db0ad79aR59-R60
[2]: https://github.com/Chiel92/vim-autoformat/issues/170
